File: psets/pset2/readability.c
// Program which classifies a sentence entered by the user according to its Coleman-Liau index
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// find numbers of letters, sentences and words in the text
int get_letters(char* text);
int get_sentences(char* text);
int get_words(char* text);

int main(void)
{
    // ask user for text and store it in the text variable
    char text[50];
	printf("Text: ");
	scanf("%s", text);
    // storing number of letters, sentences and words in text in individual variables
    int letters = get_letters(text);
    int sentences = get_sentences(text);
    int words = get_words(text);
    // get letters and sentences per 100 words
    float letters100 = ((float) letters / (float) words) * 100;
    float sentences100 = ((float) sentences / (float) words) * 100;
    // calculate the grade level readability
    int grade = round(0.0588 * letters100 - 0.296 * sentences100 - 15.8);
    // print your findings
    if(grade < 1)
    {
        printf("Before Grade 1...\n");
    }
    else if(grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int get_letters(char* text)
{
    int total = 0;
    int i;
    for(i = 0; i < strlen(text); i++)
    {
        if(isupper(text[i]) || islower(text[i]))
        {
            total += 1;
        }
    }
    return total;
}

int get_sentences(char* text)
{
    int total = 0;
    int i;
    for(i = 0; i < strlen(text); i++)
    {
        if(text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            total += 1;
        }
    }
    return total;
}

int get_words(char* text)
{
    int total = 1;
    int i;
    for(i = 0; i < strlen(text); i++)
    {
        if(text[i] == ' ')
        {
            total += 1;
        }
    }
    return total;
}

File: psets/pset6/readability.py
# Program which classifies a sentence entered by the user according to its Coleman-Liau index
# The Coleman-Liau index is computed as 0.0588 * L - 0.296 * S - 15.8,
# where L is the average number of letters per 100 words in the text,
# and S is the average number of sentences per 100 words in the text

def main():
    sentence = get_input()

    # Find number of letters
    num_letters = 0
    for c in sentence:
        if(c.isalpha()):
            num_letters += 1

    # Find number of words
    num_words = sentence.count(" ") + 1

    # Find number of sentences
    num_sentences = sentence.count(".") + sentence.count("!") + sentence.count("?")

    # Average number of letters per 100 words
    L = num_letters * (100 / num_words)

    # Average number of sentences per 100 words
    S = num_sentences * (100 / num_words)

    # Calculate grade
    grade = round(0.0588 * L - 0.296 * S - 15.8)

    # Display grade to user
    if grade < 1:
        print("Before Grade 1")
    elif(grade >= 16):
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def get_input():
    sentence = input("Text: ")
    return sentence



main()